### ISSUE TITLE
Minimize cost for March 2020 release

### ISSUE THREAD

**Issue by hannes-ucsc** [DataBiosphere/cellxgene-fargate#7]
2bfe58df6d0c3b5b3f42d03d01df5954736735b5 crudely derives the container memory from the matrix size. 2G are probably overkill for the smaller matrices.

Optimize the allocated container memory for each matrix. Determine the impact of backed mode on this and potentially enable it.

Furthermore, we can tune the cost vs performance knob by tweaking the vCPU allocated for each container. Note that memory can't be freely chosen: only increments of 1G are allowed and the allowed range depends on the vCPU: https://aws.amazon.com/fargate/pricing/

Note that fractional vCPU is permitted, down to a quarter of a vCPU.

As far as I understand that page, the pre-dominant cost factor appears to be vCPU, not memory so our strategy should be to choose the lowest vCPU that includes the minimum RAM requirement for a matrix as long as the end-user experience is acceptable preformance-wise. 

There should be an automated way to estimate the hourly cost of the entire cluster. Either the cost could be logged when the TF JSON is created from the template (make config) or we could try to use a TF output variable so the estimated cost is printed at the end of `make apply`. I think the former is more convenient because you don't need to do a risky `make plan` to get that info. Then again, a log statement emitted during `make config` scrolls by fairly quickly. Maybe we should do both, with the log statement being the MVP.

[edit: added last paragraph]

**Comment by hannes-ucsc** [comment]
@noah-aviel-dove, aaa3f83 make the sizing less crude but there's still a lot of work to do. I've only tested whether each site loads, I haven't tried actually using the app for analyses. 

I also modified the health check to allow 10min for the app to load (4a15a67b6fc405613af8e96f363d0328d036241a). Be sure you understand how the LB health checks interoperate with how Fargate schedules replacement containers. With the default health checks (1min before the target is considered unhealthy) the three largest container were killed while they were still loading. Then again, 10min may be too long. Also the timeout of 30s may be too long. We need to figure out what a realistic response time is for the / path. I've always seen it come back in under a second. I just wanted to be on the safe side for now, to prevent containers from being killed to eagerly by the HC.

Also note the added last paragraph in the description.
